internal/resources/slo/data_source_slo.go

```diff
@@ -99,6 +100,22 @@ func unpackQuery(apiquery gapi.Query) []map[string]interface{} {
 		retQuery = append(retQuery, query)
 	}
 
+	if apiquery.Type == gapi.QueryTypeRatio {
+		query := map[string]interface{}{"type": gapi.QueryTypeRatio}
+
+		ratio := []map[string]interface{}{}
+		body := map[string]interface{}{
+			"success_metric":  apiquery.Ratio.SuccessMetric.PrometheusMetric,
+			"total_metric":    apiquery.Ratio.TotalMetric.PrometheusMetric,
+			"group_by_labels": apiquery.Ratio.GroupByLabels,
+		}
+
+		ratio = append(ratio, body)
+		query["ratio"] = ratio
+
+		retQuery = append(retQuery, query)
+	}
+
```

Comment:
do you need error handling for case where `apiQuery.Type` didn't match either of the if blocks?

Reply:
This is on the READ Path from our SLO API back to Terraform (not from the User Input into Terraform) so we should always have the apiQuery type being either Ratio or Freeform. 

The user also gets a validation error if they input anything other than freeform or ratio within the type on the terraform state file as well.